website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,388 @@
+```
+
+And now, we just need to unbind the texture.
+```c#
+_gl.BindTexture(TextureTarget.Texture2D, 0);
+```
+
+## Handling texture data in shader
+Now, let's finally show it on the screen!
+
+In your fragment shader, add this `uniform` declaration:
+```
+uniform sampler2D uTexture;
+```
+
+And change this line:
+```
+void main()
+{
+//  -out_color = vec4(frag_texCoords.x, frag_texCoords.y, 0, 1.0);
+    out_color = texture(uTexture, frag_texCoords);
+}
+```
+
+In GLSL, you usually dont work with textures directly. The most common way to access a texture is using a `sampler`.
+In our case, we are using `sampler2D` for our 2D texture.
+To read the texture data, the `texture` method is used. It samples the color of the texture at the
+specified texture coordinates and uses the previously set parameters to interpolate that color.
+
+But pay attention! If you have good eyes, you noticed our texture coordinates are between the ranges of `0.0` to `1.0` This is because the `texture` method works
+with normalized values! If you dont understand, think about a 250x500 pixels image. If you want to get the pixel at the position
+(250, 250), we need to send (1, 0.5) as the texture coordinate. You can derive the pixel coordinate from the normalized coordinate by multiplying by the axis' size.
+
+<?# Info "You can use the equation ` 1/size * pixel_position ` to get the normalized coordinate for a particlar axis!" /?>
+
+Now, if you run the program, you will see just a black quad. It's because we need to bind the texture before the draw call!
+If you don't do it, the last applied texture will be used for this mesh, in our case, since we have no texture bound, no texture is used.
+
+To solve it, go to your `OnRender` method and, below the `_gl.UseProgram()` call, first activate the texture slot that you will use:
+```c#
+_gl.ActiveTexture(TextureUnit.Texture0);
+```
+And after, bind the texture!
+```c#
+_gl.BindTexture(TextureTarget.Texture2D, _texture);
+```
+
+And now when you run it (drumroll...), you can see the image being drawn inside the quad!
+
+![Quad with texture](../../../images/opengl/chapter1/leasson3/quad-with-texture.png)
+
+## Texture parameters
+And now, some extra content to go with this tutorial!
+
+If you get curious about how the texture parameters work, let's learn it now!
+
+First, let's learn the structure of the command:
+```c#
+_gl.TextureParameter( [Texture ID] , [Parameter to change] , [Value to change] );
+```
+
+The texture ID you already know. So let's learn what each parameter means!
+
+### `TextureWrapS` & `TextureWrapT`:
+`TextureWrap` is the parameter that allows you to say what the `texture` method (in the fragment shader) should do with values less than 0 and
+greater than 1. The `S` and `T` refer to X and Y respectively.
+
+Let's see the most common values for these parameters:
+
+#### `TextureWrapMode.Repeat`:
+Just repeats the image without any change, like the value never got less than 0 or greater than 1.  
```

Comment:
> like the value never got less than 0 or greater than 1

This doesn't mean anything (to me, at least)

```suggestion
Just repeats the image without any change, turning any coordinates outside the [0, 1) range by taking the fractional part of said number.
```

Reply:
i should use `[0, 1]` or `(0, 1)` ? Don't answer it it doesn't matter

Reply:
AFAIK if you want to be precise, it's [0, 1).

Reply:
oh, I understand it lol i thought it was an error (made a really big time since I studied sets)